lib/config/validation-helpers/types.ts

```diff
@@ -4,3 +4,8 @@ export interface CheckManagerArgs {
   resolvedRule: PackageRule;
   currentPath: string;
 }
+
+export interface CheckMatcherArgs {
+  val: unknown[];
```

Comment:
```suggestion
  val: unknown;
```
should work too, `is.array` will narrow the type